contracts/BSCValidatorSet.sol

```diff
@@ -312,25 +354,38 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return CODE_OK;
   }
 
-  function getValidators()external view returns(address[] memory) {
+  function getValidators() public view returns (address[] memory) {
     uint n = currentValidatorSet.length;
     uint valid = 0;
     for (uint i = 0;i<n;i++) {
-      if (!currentValidatorSet[i].jailed) {
+      if (isWorkingValidator(i)) {
         valid ++;
       }
     }
     address[] memory consensusAddrs = new address[](valid);
     valid = 0;
     for (uint i = 0;i<n;i++) {
-      if (!currentValidatorSet[i].jailed) {
+      if (isWorkingValidator(i)) {
         consensusAddrs[valid] = currentValidatorSet[i].consensusAddress;
         valid ++;
       }
     }
     return consensusAddrs;
   }
-  
+
+  function isWorkingValidator(uint index) public view returns (bool) {
+    if (index >= currentValidatorSet.length) {
+      return false;
+    }
+
+    // validatorExtraSet[index] should not be used before it has been init.
+    if (index >= validatorExtraSet.length) {
+      return !currentValidatorSet[index].jailed;
+    }
+
+    return !currentValidatorSet[index].jailed && !validatorExtraSet[index].isMaintaining;
```

Comment:
As this is a public function, the external user may call getValidators() before init the extraSet, so we need protect logic here, otherwise the out of memory access will make this function not success.

Reply:
protected on line 382 , check it bro